pkg/validator/cluster.go

```diff
@@ -914,6 +973,10 @@ func admitClusterCR(
 	// metadata generation number here because that is incremented after this
 	// validator sees the request.
 	if ar.Request.Operation == v1beta1.Update {
+		normalizeRolesResourceQuantities(
+			prevClusterCR.Spec.Roles,
+			clusterCR.Spec.Roles,
+		)
 		if reflect.DeepEqual(clusterCR.Spec, prevClusterCR.Spec) {
```

Comment:
How about using equality.Semantic.DeepEqual instead? It already fixes the issue with Quantities (see https://pkg.go.dev/k8s.io/apimachinery/pkg/api/equality)

Reply:
This is CR spec comparison and not just quantity

Reply:
> This is CR spec comparison and not just quantity

Yes, I understand it. And semantic equality was designed for these kind of situations. It works same as reflect.DeepEquals but compares objects like Quantity, Time etc by their equality functions

Reply:
Neat! I'll have a look.

Reply:
If this works then it will take care of _storage_ field in role spec also. @joel-bluedata we should do the same for storage size also, right?

Reply:
Probably yes. I'm going to see if all of our uses of reflect.DeepEqual should just be equality.Semantic.DeepEqual instead.